src/neo/Trie/MPT/MPTReadOnlyTrie.Find.cs

```diff
@@ -0,0 +1,110 @@
+using Neo.IO;
+using System;
+using System.Collections.Generic;
+using static Neo.Helper;
+
+namespace Neo.Trie.MPT
+{
+    public partial class MPTReadOnlyTrie<TKey, TValue>
+        where TKey : notnull, ISerializable, new()
+        where TValue : class, ISerializable, new()
+    {
+        private byte[] Seek(ref MPTNode node, byte[] path, out MPTNode start)
+        {
+            switch (node)
+            {
+                case LeafNode leafNode:
+                    {
+                        if (path.Length == 0)
+                        {
+                            start = leafNode;
+                            return Array.Empty<byte>();
+                        }
+                        break;
+                    }
+                case HashNode hashNode:
+                    {
+                        if (hashNode.IsEmptyNode) break;
+                        var newNode = Resolve(hashNode);
+                        if (newNode is null) break;
```

Comment:
we don't set the `start` as `hashNode`?

Reply:
If it is `HashNode`, we resolve it to specific node type and use that as `start`.